### ISSUE TITLE
Logicmonitor Sensor Crashes after startup

### ISSUE THREAD

**Issue by setswei** [StackStorm-Exchange/stackstorm-logicmonitor#7]
Hey team.

the logicmonitor sensor is crashing after it starts up straight away.

```

2023-10-29 01:50:57,131 140046079595968 INFO process_container [-] Process for sensor logicmonitor.LogicMonitorSensor has exited with code 1
2023-10-29 01:53:05,929 140046140560832 INFO process_container [-] Container shutting down. Invoking cleanup on sensors.
2023-10-29 01:53:06,931 140046140560832 INFO process_container [-] All sensors are shut down.
2023-10-29 01:53:06,955 140046140560832 INFO manager [-] (PID:186236) SensorContainer stopped. Reason - SystemExit
2023-10-29 01:53:15,718 140327648941504 INFO __init__ [-] Connecting to database "st2" @ "127.0.0.1:27017" as user "None".
2023-10-29 01:53:15,773 140327648941504 INFO __init__ [-] Successfully connected to database "st2" @ "127.0.0.1:27017" as user "None".
2023-10-29 01:53:17,882 140327648941504 INFO partitioner_lookup [-] Using partitioner default with sensornode sensornode1.
2023-10-29 01:53:17,902 140327648941504 INFO partitioners [-] Found 2 registered sensors in db scan.
2023-10-29 01:53:17,906 140327648941504 INFO manager [-] Setting up container to run 2 sensors.
2023-10-29 01:53:17,916 140327648941504 INFO manager [-]        Sensors list - ['linux.FileWatchSensor', 'logicmonitor.LogicMonitorSensor'].
2023-10-29 01:53:17,918 140327648941504 INFO manager [-] (PID:186793) SensorContainer started.
2023-10-29 01:53:17,920 140327587973600 INFO process_container [-] Running sensor linux.FileWatchSensor
2023-10-29 01:53:18,014 140327587973776 INFO mixins [-] Connected to amqp://guest:**@127.0.0.1:5672//
2023-10-29 01:53:18,214 140327587973600 INFO process_container [-] Sensor linux.FileWatchSensor started
2023-10-29 01:53:18,215 140327587973600 INFO process_container [-] Running sensor logicmonitor.LogicMonitorSensor
2023-10-29 01:53:18,380 140327587973600 INFO process_container [-] Sensor logicmonitor.LogicMonitorSensor started
2023-10-29 01:53:28,075 140346341836048 INFO sensor_wrapper [-] No config found for sensor "FileWatchSensor"
2023-10-29 01:53:28,076 140346341836048 INFO sensor_wrapper [-] Watcher started
2023-10-29 01:53:28,076 140346341836048 INFO sensor_wrapper [-] Running sensor initialization code
2023-10-29 01:53:28,114 140346289367520 INFO mixins [-] Connected to amqp://guest:**@127.0.0.1:5672//
2023-10-29 01:53:28,117 140346341836048 INFO sensor_wrapper [-] Running sensor in passive mode
2023-10-29 01:53:28,183 140441126974736 ERROR (unknown file) [-] Traceback (most recent call last):

2023-10-29 01:53:28,184 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 347, in _get_sensor_instance

2023-10-29 01:53:28,186 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,186 140441126974736 ERROR (unknown file) [-] sensor_class = loader.register_plugin_class(
2023-10-29 01:53:28,186 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,187 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/util/loader.py", line 151, in register_plugin_class

2023-10-29 01:53:28,187 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,187 140441126974736 ERROR (unknown file) [-] module = imp.load_source(module_name, file_path)
2023-10-29 01:53:28,188 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,188 140441126974736 ERROR (unknown file) [-]   File "/usr/lib/python3.8/imp.py", line 171, in load_source

2023-10-29 01:53:28,189 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,189 140441126974736 ERROR (unknown file) [-] module = _load(spec)
2023-10-29 01:53:28,190 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,191 140441126974736 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 702, in _load

2023-10-29 01:53:28,191 140441126974736 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked

2023-10-29 01:53:28,191 140441126974736 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap_external>", line 848, in exec_module

2023-10-29 01:53:28,192 140441126974736 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed

2023-10-29 01:53:28,195 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py", line 4, in <module>

2023-10-29 01:53:28,196 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,196 140441126974736 ERROR (unknown file) [-] from flask import request, Flask
2023-10-29 01:53:28,196 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,198 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/__init__.py", line 7, in <module>

2023-10-29 01:53:28,198 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,198 140441126974736 ERROR (unknown file) [-] from .app import Flask as Flask
2023-10-29 01:53:28,199 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,199 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/app.py", line 28, in <module>

2023-10-29 01:53:28,199 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,201 140441126974736 ERROR (unknown file) [-] from . import cli
2023-10-29 01:53:28,202 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,202 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/cli.py", line 18, in <module>

2023-10-29 01:53:28,202 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,203 140441126974736 ERROR (unknown file) [-] from .helpers import get_debug_flag
2023-10-29 01:53:28,203 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,203 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/helpers.py", line 16, in <module>

2023-10-29 01:53:28,204 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,204 140441126974736 ERROR (unknown file) [-] from werkzeug.urls import url_quote
2023-10-29 01:53:28,204 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,205 140441126974736 ERROR (unknown file) [-] ImportError
2023-10-29 01:53:28,205 140441126974736 ERROR (unknown file) [-] :
2023-10-29 01:53:28,205 140441126974736 ERROR (unknown file) [-] cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)
2023-10-29 01:53:28,205 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,206 140441126974736 ERROR (unknown file) [-]
During handling of the above exception, another exception occurred:


2023-10-29 01:53:28,206 140441126974736 ERROR (unknown file) [-] Traceback (most recent call last):

2023-10-29 01:53:28,207 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 441, in <module>

2023-10-29 01:53:28,207 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,209 140441126974736 ERROR (unknown file) [-] obj = SensorWrapper(
2023-10-29 01:53:28,210 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,210 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 264, in __init__

2023-10-29 01:53:28,210 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,211 140441126974736 ERROR (unknown file) [-] self._sensor_instance = self._get_sensor_instance()
2023-10-29 01:53:28,211 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,211 140441126974736 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 361, in _get_sensor_instance

2023-10-29 01:53:28,212 140441126974736 ERROR (unknown file) [-]
2023-10-29 01:53:28,212 140441126974736 ERROR (unknown file) [-] raise exc_cls(msg)
2023-10-29 01:53:28,212 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,213 140441126974736 ERROR (unknown file) [-] ImportError
2023-10-29 01:53:28,213 140441126974736 ERROR (unknown file) [-] :
2023-10-29 01:53:28,213 140441126974736 ERROR (unknown file) [-] Failed to load sensor class from file "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py" (sensor file most likelydoesn't exist or contains invalid syntax): cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)

Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 347, in _get_sensor_instance
    sensor_class = loader.register_plugin_class(
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/util/loader.py", line 151, in register_plugin_class
    module = imp.load_source(module_name, file_path)
  File "/usr/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py", line 4, in <module>
    from flask import request, Flask
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/__init__.py", line 7, in <module>
    from .app import Flask as Flask
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/app.py", line 28, in <module>
    from . import cli
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/cli.py", line 18, in <module>
    from .helpers import get_debug_flag
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/helpers.py", line 16, in <module>
    from werkzeug.urls import url_quote
ImportError: cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)

2023-10-29 01:53:28,215 140441126974736 ERROR (unknown file) [-]

2023-10-29 01:53:28,398 140327587973600 INFO process_container [-] Process for sensor logicmonitor.LogicMonitorSensor has exited with code 1
2023-10-29 01:53:32,864 140483908572432 ERROR (unknown file) [-] Traceback (most recent call last):

2023-10-29 01:53:32,865 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 347, in _get_sensor_instance

2023-10-29 01:53:32,865 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,865 140483908572432 ERROR (unknown file) [-] sensor_class = loader.register_plugin_class(
2023-10-29 01:53:32,866 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,866 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/util/loader.py", line 151, in register_plugin_class

2023-10-29 01:53:32,866 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,867 140483908572432 ERROR (unknown file) [-] module = imp.load_source(module_name, file_path)
2023-10-29 01:53:32,867 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,867 140483908572432 ERROR (unknown file) [-]   File "/usr/lib/python3.8/imp.py", line 171, in load_source

2023-10-29 01:53:32,867 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,868 140483908572432 ERROR (unknown file) [-] module = _load(spec)
2023-10-29 01:53:32,868 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,868 140483908572432 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 702, in _load

2023-10-29 01:53:32,869 140483908572432 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked

2023-10-29 01:53:32,869 140483908572432 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap_external>", line 848, in exec_module

2023-10-29 01:53:32,869 140483908572432 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed

2023-10-29 01:53:32,869 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py", line 4, in <module>

2023-10-29 01:53:32,870 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,870 140483908572432 ERROR (unknown file) [-] from flask import request, Flask
2023-10-29 01:53:32,870 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,871 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/__init__.py", line 7, in <module>

2023-10-29 01:53:32,871 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,871 140483908572432 ERROR (unknown file) [-] from .app import Flask as Flask
2023-10-29 01:53:32,872 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,872 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/app.py", line 28, in <module>

2023-10-29 01:53:32,872 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,872 140483908572432 ERROR (unknown file) [-] from . import cli
2023-10-29 01:53:32,873 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,873 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/cli.py", line 18, in <module>

2023-10-29 01:53:32,873 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,873 140483908572432 ERROR (unknown file) [-] from .helpers import get_debug_flag
2023-10-29 01:53:32,874 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,874 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/helpers.py", line 16, in <module>

2023-10-29 01:53:32,875 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,875 140483908572432 ERROR (unknown file) [-] from werkzeug.urls import url_quote
2023-10-29 01:53:32,885 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,885 140483908572432 ERROR (unknown file) [-] ImportError
2023-10-29 01:53:32,885 140483908572432 ERROR (unknown file) [-] :
2023-10-29 01:53:32,885 140483908572432 ERROR (unknown file) [-] cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)
2023-10-29 01:53:32,885 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,886 140483908572432 ERROR (unknown file) [-]
During handling of the above exception, another exception occurred:


2023-10-29 01:53:32,886 140483908572432 ERROR (unknown file) [-] Traceback (most recent call last):

2023-10-29 01:53:32,886 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 441, in <module>

2023-10-29 01:53:32,886 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,887 140483908572432 ERROR (unknown file) [-] obj = SensorWrapper(
2023-10-29 01:53:32,887 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,887 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 264, in __init__

2023-10-29 01:53:32,887 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,887 140483908572432 ERROR (unknown file) [-] self._sensor_instance = self._get_sensor_instance()
2023-10-29 01:53:32,887 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,887 140483908572432 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 361, in _get_sensor_instance

2023-10-29 01:53:32,888 140483908572432 ERROR (unknown file) [-]
2023-10-29 01:53:32,888 140483908572432 ERROR (unknown file) [-] raise exc_cls(msg)
2023-10-29 01:53:32,888 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:32,888 140483908572432 ERROR (unknown file) [-] ImportError
2023-10-29 01:53:32,888 140483908572432 ERROR (unknown file) [-] :
2023-10-29 01:53:32,888 140483908572432 ERROR (unknown file) [-] Failed to load sensor class from file "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py" (sensor file most likelydoesn't exist or contains invalid syntax): cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)

Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 347, in _get_sensor_instance
    sensor_class = loader.register_plugin_class(
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/util/loader.py", line 151, in register_plugin_class
    module = imp.load_source(module_name, file_path)
  File "/usr/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py", line 4, in <module>
    from flask import request, Flask
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/__init__.py", line 7, in <module>
    from .app import Flask as Flask
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/app.py", line 28, in <module>
    from . import cli
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/cli.py", line 18, in <module>
    from .helpers import get_debug_flag
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/helpers.py", line 16, in <module>
    from werkzeug.urls import url_quote
ImportError: cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)

2023-10-29 01:53:32,888 140483908572432 ERROR (unknown file) [-]

2023-10-29 01:53:33,403 140327587973600 INFO process_container [-] Process for sensor logicmonitor.LogicMonitorSensor has exited with code 1
2023-10-29 01:53:40,569 140562305545488 ERROR (unknown file) [-] Traceback (most recent call last):

2023-10-29 01:53:40,570 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 347, in _get_sensor_instance

2023-10-29 01:53:40,571 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,572 140562305545488 ERROR (unknown file) [-] sensor_class = loader.register_plugin_class(
2023-10-29 01:53:40,573 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,573 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/util/loader.py", line 151, in register_plugin_class

2023-10-29 01:53:40,574 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,574 140562305545488 ERROR (unknown file) [-] module = imp.load_source(module_name, file_path)
2023-10-29 01:53:40,575 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,576 140562305545488 ERROR (unknown file) [-]   File "/usr/lib/python3.8/imp.py", line 171, in load_source

2023-10-29 01:53:40,576 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,577 140562305545488 ERROR (unknown file) [-] module = _load(spec)
2023-10-29 01:53:40,577 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,578 140562305545488 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 702, in _load

2023-10-29 01:53:40,578 140562305545488 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked

2023-10-29 01:53:40,579 140562305545488 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap_external>", line 848, in exec_module

2023-10-29 01:53:40,580 140562305545488 ERROR (unknown file) [-]   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed

2023-10-29 01:53:40,580 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py", line 4, in <module>

2023-10-29 01:53:40,583 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,584 140562305545488 ERROR (unknown file) [-] from flask import request, Flask
2023-10-29 01:53:40,584 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,585 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/__init__.py", line 7, in <module>

2023-10-29 01:53:40,586 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,586 140562305545488 ERROR (unknown file) [-] from .app import Flask as Flask
2023-10-29 01:53:40,587 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,587 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/app.py", line 28, in <module>

2023-10-29 01:53:40,588 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,588 140562305545488 ERROR (unknown file) [-] from . import cli
2023-10-29 01:53:40,589 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,590 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/cli.py", line 18, in <module>

2023-10-29 01:53:40,590 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,591 140562305545488 ERROR (unknown file) [-] from .helpers import get_debug_flag
2023-10-29 01:53:40,591 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,591 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/helpers.py", line 16, in <module>

2023-10-29 01:53:40,592 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,592 140562305545488 ERROR (unknown file) [-] from werkzeug.urls import url_quote
2023-10-29 01:53:40,594 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,595 140562305545488 ERROR (unknown file) [-] ImportError
2023-10-29 01:53:40,596 140562305545488 ERROR (unknown file) [-] :
2023-10-29 01:53:40,596 140562305545488 ERROR (unknown file) [-] cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)
2023-10-29 01:53:40,597 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,597 140562305545488 ERROR (unknown file) [-]
During handling of the above exception, another exception occurred:


2023-10-29 01:53:40,598 140562305545488 ERROR (unknown file) [-] Traceback (most recent call last):

2023-10-29 01:53:40,598 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 441, in <module>

2023-10-29 01:53:40,599 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,600 140562305545488 ERROR (unknown file) [-] obj = SensorWrapper(
2023-10-29 01:53:40,601 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,602 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 264, in __init__

2023-10-29 01:53:40,603 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,604 140562305545488 ERROR (unknown file) [-] self._sensor_instance = self._get_sensor_instance()
2023-10-29 01:53:40,604 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,605 140562305545488 ERROR (unknown file) [-]   File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 361, in _get_sensor_instance

2023-10-29 01:53:40,605 140562305545488 ERROR (unknown file) [-]
2023-10-29 01:53:40,606 140562305545488 ERROR (unknown file) [-] raise exc_cls(msg)
2023-10-29 01:53:40,606 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:40,607 140562305545488 ERROR (unknown file) [-] ImportError
2023-10-29 01:53:40,607 140562305545488 ERROR (unknown file) [-] :
2023-10-29 01:53:40,607 140562305545488 ERROR (unknown file) [-] Failed to load sensor class from file "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py" (sensor file most likelydoesn't exist or contains invalid syntax): cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)

Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2reactor/container/sensor_wrapper.py", line 347, in _get_sensor_instance
    sensor_class = loader.register_plugin_class(
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/util/loader.py", line 151, in register_plugin_class
    module = imp.load_source(module_name, file_path)
  File "/usr/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/opt/stackstorm/packs/logicmonitor/sensors/logicmonitor_sensor.py", line 4, in <module>
    from flask import request, Flask
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/__init__.py", line 7, in <module>
    from .app import Flask as Flask
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/app.py", line 28, in <module>
    from . import cli
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/cli.py", line 18, in <module>
    from .helpers import get_debug_flag
  File "/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/flask/helpers.py", line 16, in <module>
    from werkzeug.urls import url_quote
ImportError: cannot import name 'url_quote' from 'werkzeug.urls' (/opt/stackstorm/virtualenvs/logicmonitor/lib/python3.8/site-packages/werkzeug/urls.py)

2023-10-29 01:53:40,610 140562305545488 ERROR (unknown file) [-]

2023-10-29 01:53:43,413 140327587973600 INFO process_container [-] Process for sensor logicmonitor.LogicMonitorSensor has exited with code 1
```

has anyone seen this before is is something now broken in this pack

**Comment by setswei** [comment]
This seems to be caused by an issue with the latest release of werkzeug URL library.  adding werkzeug==2.2.2 to the requirements.txt file resolves the issues

**Comment by arm4b** [comment]
@setswei Good find! Surely open a PR to fix it.

**Comment by setswei** [comment]
I have created a pull request with the fix ( PR #8)

**Comment by setswei** [comment]
I will have to fix the build as well it is using 3.6 which werkzeug==2.2.2 is not available

**Comment by setswei** [comment]
Hey @armab 

I have tested the CI pipeline using Python 3.8 and it works

![image](https://github.com/StackStorm-Exchange/stackstorm-logicmonitor/assets/7195599/5552f36d-37e3-418e-8448-93616942e72c)

There is an open pull request [PR #138](https://github.com/StackStorm-Exchange/ci/pull/138) that needs to be approved. Is there i can do to assist with getting that one merged ?
